epsagon/utils.py

```diff
@@ -445,3 +456,123 @@ def database_connection_type(hostname, default_type):
     if 'redshift.amazonaws' in hostname:
         return 'redshift'
     return default_type
+
+
+def build_split_string(options):
+    """
+    returns regex string to split on option(s)
+    :param options: Union[str: 1 option, list[str]: multiple options]
+    :param options: options to split string on
+    :return: string type, compatible for multiple options of regex split
+    """
+    if isinstance(options, list):
+        options = '|'.join(map(re.escape, options))
+
+    # if options is not a string by now param was invalid
+    if not isinstance(options, str):
+        return ''
+
+    return '({options})'.format(options=options)
+
+
+def obfuscate_sql_query(query, operation):
+    """
+    Obfuscate SQL queries to protect sensitive uploads
+    :param query: The SQL query string
+    :param operation: Operation in DB
+    :return: the obfuscated query (string)
+    :var bounds: specifier for locating values based upon operation
+    :var bounds: token_type is sqlparse type
+    :var bounds: leftmost bound and rightmost bound are absolute limits
+    :var bounds: separator is the symbol(s) between values
+    :var bounds: signal is a preceding identifier for each value
+    """
+    bounds = {
+        'select': [
+            ('token_type', sqlparse.sql.Where),
+            ('left_bound', 'WHERE '),
+            ('right_bound', None),
+            ('separators', [')', ' AND ', ' OR ']),
+            ('operators', ['>=', '<=', '>', '<', '=', '<>'])
+        ],
+        'insert': [
+            ('token_type', sqlparse.sql.Values),
+            ('left_bound', 'VALUES ('),
+            ('right_bound', ')'),
+            ('separators', ', '),
+            ('operators', None),
+        ],
+
+    }
+
+    if isinstance(query, bytes):
+        query = query.decode('UTF-8')
+
+    if isinstance(operation, bytes):
+        operation = operation.decode('UTF-8')
+
+    if operation not in bounds.keys():
+        return query
+
+    # order not preserved in py2. Dont default to OrderedDict,not inefficient
+    to_dict = collections.OrderedDict if sys.version_info.major == 2 else dict
+
+    [token_type,
+    start, stop,
+    separators, operators] = to_dict(bounds[operation]).values()
+    obfuscated_query = []
+
+    parse_query = sqlparse.parse(query)[0].tokens
```

Comment:
it might be possible the query will include more than one statement (returned from sqlparse)
In that case, make sure to skip the query (and print_debug about it)